tests/test_custom_models.py

```diff
@@ -444,6 +457,128 @@ def test_repr_lora_conv2d(self):
         self.assertTrue("default" in print_output)
 
 
+class MultipleActiveAdaptersTester(unittest.TestCase):
+    """
+    A test class to test the functionality of multiple active adapters.
+
+    This is not specifically tied to custom models, it's just easy to test here and testing it on all types of models
+    would be overkill.
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.configs = {
+            "lora": {"class": LoraConfig, "kwargs": {"target_modules": ["lin0"], "init_lora_weights": False}},
+            "ia3": {
+                "class": IA3Config,
+                "kwargs": {
+                    "target_modules": ["lin0"],
+                    "feedforward_modules": ["lin0"],
+                    "init_ia3_weights": False,
+                },
+            },
+            "adalora": {
+                "class": AdaLoraConfig,
+                "kwargs": {"target_modules": ["lin0"], "init_lora_weights": False, "inference_mode": True},
+            },
+        }
+
+    def prepare_inputs_for_testing(self):
+        X = torch.arange(90).view(9, 10)
+        return {"X": X}
+
+    def get_adapter_config(self, tuner_method, targets):
+        if targets == "same":
+            return self.configs[tuner_method]
+        config = copy.deepcopy(self.configs[tuner_method])
+        kwargs = config["kwargs"]
+        for key, value in kwargs.items():
```

Comment:
Hmm, I need 2 configs per test. The way `TEST_CASES` is available, it can't be used to get 2 configs per test.

Reply:
So I imagine it could look something like this:

```python
MULTIPLE_ACTIVE_ADAPTERS_CONFIGS = {
    "lora": (
        LoraConfig(target_modules=["lin0"], init_lora_weights=False),
        LoraConfig(target_modules=["lin0", "lin1"], init_lora_weights=False),
    ),
    "ia3": (
        IA3Config(target_modules=["lin0"], feedforward_modules=["lin0"], init_ia3_weights=False),
        IA3Config(target_modules=["lin0", "lin1"], feedforward_modules=["lin0", "lin1"], init_ia3_weights=False),
    ),
    # etc.
}


class MultipleActiveAdaptersTester(unittest.TestCase):
    # ...
    @parameterized.expand(MULTIPLE_ACTIVE_ADAPTERS_CONFIGS.keys())
    def test_multiple_active_adapters_forward(self, tuner_method):
        config_1, config_2 = MULTIPLE_ACTIVE_ADAPTERS_CONFIGS[tuner_method]
        model = MLP(bias=tuner_method != "ia3")
        peft_model = get_peft_model(model, config_1, adapter_name="adapter_1")
        peft_model.add_adapter("adapter_2", config_2)
        ...
```

Reply:
addressed in the latest commit